dashboard/app/models/pd/application/application_base.rb

```diff
@@ -130,6 +130,19 @@ def set_type_and_year
     validates_presence_of :type
     validates_presence_of :status, unless: proc {|application| application.application_type == PRINCIPAL_APPROVAL_APPLICATION}
 
+    # decision notifications should be sent to applications that have been
+    # locked but have not yet received a decision_notification email
+    scope :should_send_decision_notification_emails, -> {where("locked_at IS NOT NULL AND decision_notification_email_sent_at IS NULL")}
+
+    # Override in derived class
+    def send_decision_notification_email
+      raise 'Abstract method must be overridden by inheriting class'
```

Comment:
If we add a new application type that can be locked, this will start failing the mailer. It might be better to remove the `raise` and just have the base implementation be a no-op.